src/module.c

```diff
@@ -7680,6 +7884,37 @@ void moduleHandleBlockedClients(void) {
             reply_us = elapsedUs(replyTimer);
             moduleFreeContext(&ctx);
         }
+        /* Call the custom auth reply cb if one exists if custom auth is in progress. */
+        RedisModuleCustomAuthCtx *auth_ctx = bc->auth_ctx;
+        if (c && c->flags & CLIENT_CUSTOM_AUTH && !bc->blocked_on_keys && bc->auth_reply_cb) {
+            /* Reset the client's CUSTOM_AUTH_RESULT flag before attempting the auth_reply_cb. */
+            c->flags &= ~CLIENT_CUSTOM_AUTH_RESULT;
+            robj *username = NULL;
+            robj *password = NULL;
+            long long ver = 0;
+            robj *clientname = NULL;
+            int is_auth_cmd = getAuthOrHelloAuthCmdArgs(c, &username, &password, &ver, &clientname);
```

Comment:
I don't think you fully got my previous point about reprocessing the command. The idea was that if the auth and hello commands were fully re-entrant, we would be able to keep calling them over and over again until they succeeded without all this back code. Let me know if my comment still isn't clear.